src/explore-education-statistics-admin/src/components/editable/EditableKeyStat.tsx

```diff
@@ -1,205 +1,81 @@
-            });
-            toggleShowForm.off();
-          }}
-        >
-          {form => (
-            <Form id={formId}>
-              <h3 className="govuk-heading-s">{name}</h3>
-
-              <KeyStatTile
-                title={keyStat.title}
-                titleTag="h4"
-                testId={testId}
-                value={keyStat.value}
-                isReordering={isReordering}
-              >
-                <FormFieldTextInput<KeyStatsFormValues>
-                  name="dataSummary"
-                  label={<span className={styles.trendText}>Trend</span>}
-                />
-              </KeyStatTile>
-
-              <FormFieldTextInput<KeyStatsFormValues>
-                formGroupClass="govuk-!-margin-top-2"
-                name="dataDefinitionTitle"
-                label="Guidance title"
-              />
-
-              <FormFieldEditor<KeyStatsFormValues>
-                name="dataDefinition"
-                toolbarConfig={toolbarConfigs.simple}
-                label="Guidance text"
-              />
-
-              <ButtonGroup>
-                <Button
-                  disabled={!form.isValid}
-                  type="submit"
-                  className="govuk-!-margin-right-2"
-                >
-                  Save
-                </Button>
-                <Button variant="secondary" onClick={toggleShowForm.off}>
-                  Cancel
-                </Button>
-              </ButtonGroup>
-            </Form>
-          )}
-        </Formik>
-      );
-    }
-
-    const dataDefinitionTitle = summary?.dataDefinitionTitle[0] || 'Help';
-
+  const { updateKeyStatisticDataBlock } = useReleaseContentActions();
+  if ((keyStat as KeyStatisticDataBlock).dataBlockId) {
     return (
-      <>
-        <KeyStatTile
-          title={keyStat.title}
-          value={keyStat.value}
-          testId={testId}
-          isReordering={isReordering}
-        >
-          {summary?.dataSummary[0] && (
-            <p className="govuk-body-s" data-testid={`${testId}-summary`}>
-              {summary.dataSummary[0]}
-            </p>
-          )}
-        </KeyStatTile>
-
-        {summary?.dataDefinition[0] && !isReordering && (
-          <Details
-            summary={dataDefinitionTitle}
-            className={styles.definition}
-            hiddenText={
-              dataDefinitionTitle === 'Help'
-                ? `for ${keyStat.title}`
-                : undefined
-            }
-          >
-            <div data-testid={`${testId}-definition`}>
-              {summary.dataDefinition.map(data => (
-                <ReactMarkdown key={data}>{data}</ReactMarkdown>
-              ))}
-            </div>
-          </Details>
-        )}
-
-        {isEditing && !isReordering && (
-          <ButtonGroup className="govuk-!-margin-top-2">
-            <Button onClick={toggleShowForm.on}>Edit</Button>
+      <EditableKeyStatDataBlock
+        keyStat={keyStat as KeyStatisticDataBlock}
```

Comment:
I wonder if you can get this type casting for free earlier on if there's some form of type guard you can make use of.

Reply:
I'm leaving this for Nick as I'm not sure what The Way is here.

Reply:
:+1: